### ISSUE TITLE
Resetting the previous screen's state works only once

### ISSUE THREAD

**Issue by mica16** [react-community/react-navigation#1300]
_Using the Git master's version and react-native 0.43._

Basically I have those defined screens: 

```
const myNavigator = StackNavigator({
    Members: { screen: MembersList },
    Chat: { screen: Chat }
});
```

My expected behavior is the following: 

1. Opening the app, Members' list page by default, loading 300 members.
2. Clicking on one member in the list open the Chat page. 
3. Clicking on back button should reset the whole MembersList component and therefore reload those 300 members (bad UX experience to force a refresh but I expect it since it's a conceptual example).
4. Clicking again on any member in the list to reach Chat page.
5. Clicking back button should AGAIN reset the Members' list state.

I managed to make the steps from 1 to 3 work with the following code within **Chat component** (I'm using Typescript):

```
export default class Chat extends Component<Props, State> {

    static navigationOptions = ({ navigation }) => ({
        title: 'Chat',
        headerLeft: (<Button title='click' onPress={() => Chat.goBack(navigation)} />)
    });

    private static goBack(navigation) {
        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Members'})]
        });
        navigation.dispatch(resetAction);
    }

    //........
```
However, **in the step 5, state is not resetted**. 
I struggle to figure out the reason. 

To sum up, my goal is to avoid caching a particular React component associated to a route. 

Is it a known bug or did I misuse the reset feature?
Has anyone experienced it?

**Comment by mica16** [comment]
I noticed through the logs that the first "key" of Members' route is Init (when the app is launched).
Then, after each reset it goes to the same Init0 each time, maybe explaining this caching...
Weird...

**Comment by phuochau** [comment]
@mica16 do you know how to fix it? I'm having same issue right now... 😃

**Comment by phuochau** [comment]
I found the way to fix it by update the `navReducer`
```javascript
const navReducer = (state, action) => {
    console.log('navReducer', state, action);
    try {
       ...
      if (action.type === 'Navigation/RESET') {
        const navAction = action.actions[0];
        return {
          ...state,
          routes: [{
            routeName: navAction.routeName,
            params: navAction.params,
            key: Utilities.uniqueId(), // write your function to generate route it
          }],
          index: 0,
        };
      }

      newState = AppNavigator.router.getStateForAction(action, state);
      return newState || state;
    } catch (e) {
      console.log('nav reducer error', e);
      return state;
    }
  };
```

**Comment by rsavignon** [comment]
Hi  @phuochau and @mica16, I am facing the same issue. Is possible to work around it without redux integration?

thanks in advance.

**Comment by phuochau** [comment]
@rsavignon just tried react-navigation the first time for redux. The problem is when reseting the route, the route key is always `Init0`, so without redux, you just find the way to generate new route key for every RESET.

**Comment by kelset** [comment]
Pinging OP @mica16 since this issue is quite old and related to an old version of the lib. If no answer I'll close in 7 days.

**Comment by kelset** [comment]
Hi there @mica16,

In an effort to cleanup this project and prioritize a bit, since this issue had no follow up since my last comment I'm going to close it.

If you are still having the issue (especially if it's a bug report) please open a new issue that uses the new Issue Template to provide some more details to help us solve it.
